src/middlewared/middlewared/plugins/dns_client.py

```diff
@@ -57,6 +61,7 @@ async def resolve_name(self, name, rdtype, options):
         ),
         Ref('query-filters'),
         Ref('query-options'),
+        Str('throw_exception_on', default='host_failure', enum=['never', 'any_failure', 'host_failure', 'all_failure']),
```

Comment:
We typically make enums all caps, e.g. NEVER. I believe our convention is `raise_error` for cases where caller can control exception handling.